src/ResourceTile/ResourceTile.tsx

```diff
@@ -36,96 +30,82 @@ export const CssClasses = {
   ADDITIONAL_INFO: "ResourceTile--additionalInfo",
 };
 
-const SizeCssClasses = {
-  [IconSize.SMALL]: "ResourceTile--size--small",
-  [IconSize.MEDIUM]: "ResourceTile--size--medium",
-  [IconSize.LARGE]: "ResourceTile--size--large",
+const SizeCssClasses: Record<IconSize, string> = {
+  small: "ResourceTile--size--small",
+  medium: "ResourceTile--size--medium",
+  large: "ResourceTile--size--large",
 };
 
-const IconOrientationCssClasses = {
-  [IconOrientation.LANDSCAPE]: "ResourceTile--icon--landscape",
-  [IconOrientation.PORTRAIT]: "ResourceTile--icon--portrait",
-  [IconOrientation.UNKNOWN]: "",
+const IconOrientationCssClasses: Record<IconOrientation, string> = {
+  landscape: "ResourceTile--icon--landscape",
+  portrait: "ResourceTile--icon--portrait",
+  unknown: "",
 };
 
-interface BasicProps {
-  icon: string;
-  title: string;
-  url: string;
+export interface Action {
+  content: React.ReactNode;
+  href?: string;
+  onClick?: () => void;
 }
 
-/**
- * These props may be passed through several component layers, for example:
- * ResourceLinkOnDistrictPage --> AppLinkOnDistrictPage --> AppLink --> ResourceLink
- */
-export interface PassThroughProps {
-  actions?: { content: React.ReactNode; href?: string; onClick?: () => void }[];
+export interface Props {
+  actions: [] | [Action] | [Action, Action];
   additionalInfo?: React.ReactNode;
   className?: string;
   draggable?: boolean;
+  icon: string;
+  loadingSpinner: React.ReactNode;
   notes?: string;
   notify?: boolean;
   onClick?: (event?: React.MouseEvent) => void;
+  openInSameTab: boolean;
   overlays?: React.ReactNode[];
-  size?: IconSize;
-}
-
-export function extractPassThroughProps(props: any): PassThroughProps {
-  return _.pick(props, [
-    "actions",
-    "additionalInfo",
-    "draggable",
-    "className",
-    "notes",
-    "notify",
-    "onClick",
-    "overlays",
-    "size",
-  ]);
+  size: IconSize;
+  title: string;
+  url: string;
 }
 
-type Props = BasicProps & PassThroughProps;
-
 interface State {
   iconOrientation: IconOrientation;
 }
 
 export class ResourceTile extends React.PureComponent<Props, State> {
-  static defaultProps = {
+  static defaultProps: Pick<Props, "actions" | "size" | "openInSameTab"> = {
```

Comment:
TypeScript support for `defaultProps` is actually really cool! For example, we make `actions` required in the `Props` above so that when we use `actions` in this component we never have to worry about it being undefined (which is more relevant if we had strict mode on). But even though `actions` is required in the Props, TypeScript treats them as optional when actually using this component since it knows it's in `defaultProps`!

Reply:
Nice!